inst/tests/tests.Rraw

```diff
@@ -18292,3 +18292,6 @@ DT = data.table(A=1:3, key="A")
 test(2223.1, DT[.(4), nomatch=FALSE], data.table(A=integer(), key="A"))
 test(2223.2, DT[.(4), nomatch=NA_character_], data.table(A=4L, key="A"))
 
+# groupingsets by named by argument
+test(2224.1, groupingsets(data.table(iris), j = mean(Sepal.Length), by = c('Sp'='Species'), sets = list('Species')), as.data.table(iris)[, mean(Sepal.Length), Species])
```

Comment:
These are just few rows so I would add fixed objects to compare to, and not another data.table query.